CRUDRecipeEF.BL/Services/RecipeService.cs

```diff
@@ -75,6 +75,13 @@ public async Task<string> AddRecipe(RecipeDTO recipeAddDTO)
             return recipeAddDTO.Name;
         }
 
+        public async Task UpdateRecipe (RecipeDTO recipe, string newName)
+        {
+            recipe.Name = newName;
```

Comment:
Your almost there!
Step 1. Get the entity from db using the name 
Step 2. Now you can map the new name to the old name. recipe.Name = recipeDTO.Name

Reply:
I am gonna do that with _mapper?

Reply:
You can do it manually like you have already done

Reply:
I 've changed it to this:
public async Task UpdateRecipe (RecipeDTO recipe, string recipeName)
        {

            var recipeDTO = await GetRecipeByName(recipe.Name);

            recipeDTO.Name = recipeName;

            recipe.Name = recipeDTO.Name;

            await Save();
        }
But it does not save it. I see that on the debugging mode it takes the new name

Reply:
On line 1 you are recieving a recipe not dto, also use the recipeName in the method not the recipeDto. Name

What your doing then is 2 assignments, you only need to do one assignment and thats

recipefromdb.name=recipedto.name

Reply:
You are saying fot the "var recipeDTO = await Get.....". It returns a recipeDTO. (Please tell me that it is hard and I am not that stupid

Reply:
> You are saying fot the "var recipeDTO = await Get.....". It returns a recipeDTO. (Please tell me that it is hard and I am not that stupid

I think you are just confusing some things right now it happens.
await Get... doesnt return a dto it returns a recipe. to make it easier rename the parameter recipe to recipeDTO and it might make more sense.

Reply:
But when i hover my mouse over the "recipeDTO", it says that it is a recipeDTO. How can it be a recipe?

Reply:
Heres the method: 

 /// <summary>
        ///     Finds a recipe with the specified name. Throws an exception if it doesnt exist
        /// </summary>
        /// <param name="name"></param>
        /// <returns>A recipe if found</returns>
        /// <exception cref="KeyNotFoundException"></exception>
        private async Task<Recipe> GetRecipeByNameIfExists(string name)
        {
            var recipe = await _context.Recipes.Include(i => i.Ingredients)
                .SingleOrDefaultAsync(r => r.Name.ToLower() == name.ToLower().Trim());

            if (recipe == null)
            {
                _logger.LogDebug($"Attempted to get recipe that does not exist: {name}");
                throw new KeyNotFoundException("Recipe doesnt exist");
            }

            return recipe;
        }